--- /usr/lib/qt5/qml/Sailfish/Gallery/private/ZoomableImage.qml.orig
+++ /home/defaultuser/SailFish/usr/lib/qt5/qml/Sailfish/Gallery/private/ZoomableImage.qml-new
@@ -50,10 +50,10 @@
     // pixels in size. Max texture size and Max renderbuffer size are
     // both 4096, well within, so the actual cause is unknown.
     property bool isJolla1: Screen.width == 540 && Screen.height == 960
-    largePhoto.sourceSize {
-        width: isJolla1 ? 2048 : 3264
-        height: isJolla1 ? 2048 : 3264
-    }
+    //largePhoto.sourceSize {
+    //    width: isJolla1 ? 2048 : 3264
+    //    height: isJolla1 ? 2048 : 3264
+    // }

     BrightnessContrast {
         id: adjustLevels

